frontend/src/components/auth/teacher-signup/NavigationButtons.tsx

```diff
@@ -10,16 +10,23 @@ interface NavigationButtonsProps {
   onBackClick: () => void;
   continueButtonText?: string;
   backButtonText?: string;
+  continueIsLoading?: boolean;
 }
 const NavigationButtons = ({
   onContinueClick,
   onBackClick,
   continueButtonText = "Continue",
   backButtonText = "Back",
+  continueIsLoading = false,
 }: NavigationButtonsProps): React.ReactElement => {
   return (
     <>
-      <Button onClick={onContinueClick} variant="primary" width="100%">
+      <Button
+        isLoading={continueIsLoading}
```

Comment:
nit: I'm fine with not using `ActionButton` in most cases since it won't save much in terms of readability, but in this case, using an Action button would eliminate the need for this flag! I'd suggest that refactor if you have time

Reply:
good call, i'll refactor!